classes/stageAssignment/StageAssignment.php

```diff
@@ -145,4 +145,15 @@ public function scopeWithContextId(Builder $query, ?int $contextId): Builder
                 ->where('submissions.context_id', $contextId);
         });
     }
+
+
+    /**
+     * Accessor for the userId attribute.
+     *
+     * @return int The user_id as an integer.
+     */
+    public function getUserIdAttribute(): int
```

Comment:
Do we need this method in the PR? `HasCamelCasing` trait should handle all cases of transforming between camel and snake case when Model is instantiated.

Reply:
@Vitaliy-1  Yes I also wanted to avoid adding the `getUserIdAttribute` accessor. But I had to implement this accessor because the `pluck('userId')` method does not trigger the `HasCamelCasing` transformation. This trait works for direct property access and method calls, but `pluck` directly queries the database fields, which are in snake-case (`user_id`). Without the accessor, ` pluck('userId')` returns `null` values since it doesn't recognize the camel-case attribute, this is how the original code was and caused the issue of author names being visible in double-anonymous reviews. 

I initially faced this issue when trying to use `pluck('userId')` directly, which is why I added the accessor to ensure the correct transformation.

If preferred, I can remove the accessor and revert to using the snake_case user_id like `→pluck('user_id')`, but this approach is not ideal within our codebase. 

Another alternative could be using `map()` to iterate over the collection and access the `userId` attribute  but it is less efficient than `pluck()` and introduces unnecessary overhead.

Reply:
@Hafsa-Naeem, we're still developing consistent patterns around this so glad to have a chance to hash it out here. Sorry for the delay in reviewing!

I want to keep the use of actual column names in PHP (`snake_case` like `user_id`) limited to the storage code (Eloquent model or EntityDAO, collector, repository class) for that model. Outside of that, things should be `lowerCamelCase`.

The `HasCamelCasing` attribute covers operations related to an instance of a model, but not batch operations. So external code either has to materialize the set of models (untested; details might not be 100%)...
```php
$stageAssignments = StageAssignment::withSubmissionIds([$query->assocId])
    ->withRoleIds([Role::ROLE_ID_AUTHOR])
    ->get();
$excludeUsers = $stageAssignments->map(fn($stageAssignment) => $stageAssignment->userId);
```
...or, if that's going to be too much of a performance cost, I'd suggest adding a method to `lib/pkp/classes/stageAssignment/Repository.php`, which will keep the details of the `pluck` operation encapsulated with the storage code.

I think it would be a mistake to try to get Laravel to apply the `lowerCamelCase` conventions at the level of batch operations, and it'll move us further away from normal conventions a 3rd-party dev will expect.